e2e/testdata/fn-eval/resource-deletion/starlark-delete-resource.yaml

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: fn.kpt.dev/v1alpha1
+kind: StarlarkRun
+metadata:
+  name: delete-all
```

Comment:
This function is not deleting all resources. Can we rename it to `delete-resource` ?
Also, it will be better if you delete specific resource by comparing the GVK because if orchestrator re-orders the items in the list, this logic will end up deleting wrong resource.

Reply:
Updated the starlark code to remove the resource with particular GVK.